pkgs/dart_tooling_mcp_server/test/test_harness.dart

```diff
@@ -378,3 +383,29 @@ Root rootForPath(String projectPath) =>
 final counterAppPath = p.join('test_fixtures', 'counter_app');
 
 final dartCliAppsPath = p.join('test_fixtures', 'dart_cli_app');
+
+/// A test wrapper around [LocalProcessManager] that stores commands locally
+/// instead of running them by spawning sub-processes.
+class TestProcessManager extends LocalProcessManager {
+  final commands = <String>[];
```

Comment:
nit: commandsRan?

We might eventually want to make this a stream queue or something, with some helpers like `expectCommand` but this should be fine for now.